[Wen Junhua] iP

Jh123x opened the pull request.

Commit:
Add Gradle support

jh123x2 committed:
Added Level 1

Changelog:
1. Changed flavour of duke
2. Added correct Expected output
3. Added inputs into the testcases

jh123x2 committed:
Added Task to allow tasks to be added

jh123x2 committed:
Added Task list to keep track of tasks

jh123x2 committed:
Added documentation for Task list

jh123x2 committed:
Refactoring and change flavour of duke

jh123x2 committed:
Updated Task.java

Change log:
1. Added Completed boolean
2. Added isDone, markDone, markUnDone and getStatusIcon methods

jh123x2 committed:
Update TaskList.java

Change log:
1. Added size method
2. Added markDone method

jh123x2 committed:
Added Task completion to duke

Change log:
1. Added Checks for the command

jh123x2 committed:
Updated documentation

jh123x2 committed:
Refactored displayContent function

jh123x2 committed:
Create Event class

jh123x2 committed:
Created TODO class

jh123x2 committed:
Revert "Refactored displayContent function"

This reverts commit d8426917af010e79c70865113231f34022179b35.

jh123x2 committed:
Created Deadline class

jh123x2 committed:
Added Enums to keep track of States

jh123x2 committed:
Bumped Duke to level 4

Updated Duke to level-4

jh123x2 committed:
Fixed string formatting

jh123x2 committed:
Refactoring

jh123x2 committed:
Bug fixes and renamed to Ultron

jh123x2 committed:
Update runtest.bat to match ultron

jh123x2 committed:
Updated test cases

jh123x2 committed:
Shifted parseComamnd to the different Tasks

jh123x2 committed:
Refactored States

jh123x2 committed:
Added Exceptions

jh123x2 committed:
Updated help message

jh123x2 committed:
Checked Error in converting numbers

jh123x2 committed:
Fixed more bugs in Ultron

jh123x2 committed:
Changed input and expected output

jh123x2 committed:
Added method to remove class

jh123x2 committed:
Add the delete method and refactoring

jh123x2 committed:
Bug fixes

Jh123x committed:
Renamed States.java to Command.java

Jh123x committed:
Removed unused method getList in taskList

Jh123x committed:
Renamed variables and set patterns to final

Jh123x committed:
Set list to final

Jh123x committed:
Change task to abstract class

Jh123x committed:
Added an enum class for holding exception types

Jh123x committed:
Updated UltronException to match it

Jh123x committed:
Updated ultron class to match the changes

Jh123x committed:
Updated expected.txt to match the changes

Jh123x committed:
Changed unchanged variables to final

Jh123x committed:
Refactoring of methods

Jh123x committed:
Refactoring of methods

Jh123x committed:
Added an IOException to the Exception type

Jh123x committed:
Abstracted UI class out

Jh123x committed:
Fixed CheckStyles

Jh123x committed:
Add getType method

Jh123x committed:
Add new Tasklist constructor

Jh123x committed:
Add getList method

Jh123x committed:
Add getCommand to Tasks

Jh123x committed:
Add Storage to Ultron

Jh123x committed:
Add Path creation when file is not found

Jh123x committed:
Add DIRECTORY_NOT_CREATED exception type

Jh123x committed:
Changed path where data is stored

Jh123x committed:
Updated gitignore for the data.txt file

Jh123x committed:
Add Datetime parsing

Add time parsing for date and time within overloaded constructors.

Adding the code for parsing date will allow for functionalities related to the date

Jh123x committed:
Merge pull request #1 from Jh123x/branch-Level-8

Add Datetime parsing

Jh123x committed:
Merge branch 'master' into branch-Level-7

Jh123x committed:
Merge pull request #2 from Jh123x/branch-Level-7

Add harddisk saving capabilities

Jh123x committed:
Bug fix for path creation

Bug fix for path creation

Corrected the path creation so that text can be saved correctly

Jh123x committed:
Clean up Ultron

Clean up code from Ultron

Removed unused code from ultron to make the file smaller and more readable

Jh123x committed:
Update expected value of output

Update expected value of output

This is to make sure that the testing script is able to test to see if the output is correct

Jh123x committed:
Change some arguments to final

Change some arguments to final

This is to prevent unnecessary changes inside of the code.

Jh123x committed:
Improve Syntax

Improve Syntax of Storage.java

Jh123x committed:
Reformat code

Reformat code

Removed functions which might not be used and fixed indentations

Jh123x committed:
Rename Command to TaskCommand

Rename Command to TaskCommand

This is to make room for a new class Command which helps abstract the commands away from any single class

Jh123x committed:
Add ExceptionType TOO_MUCH_ARGUMENTS

Add a new exception type TOO_MUCH_ARGUMENTS

This is to be able to catch show errors for a different type of exception

Jh123x committed:
Create abstract class Command

Create abstract class Command

Create abstract class Command to store different commands

Jh123x committed:
Create ByeCommand.java

Jh123x committed:
Create DeleteCommand.java

Jh123x committed:
Create DoneCommand.java

Jh123x committed:
Create HelpCommand.java

Jh123x committed:
Create ListCommand.java

Jh123x committed:
Create Parser class

Create Parser class

This is to abstract the methods needed for parser

Jh123x committed:
Create TaskAllocator.java

Create TaskAllocator class

To ensure efficient allocation of tasks which are to be tracked

Jh123x committed:
Change wrapper to showError

Change wrapper to showError

A better name to explain what the code does.

Jh123x committed:
Update Ultron to match the code

Update Ultron to match the code

Cleaned up Ultron to match the rest of the items

Jh123x committed:
Bug fix for data not loading correctly

Bug fix for data not loading correctly

Changed the way the data is loaded to fix the bug of data not loading correctly

Jh123x committed:
Update EXPECTED.TXT

Update EXPECTED.TXT to reflect currently expected outputs

This is to facillitate testing

Jh123x committed:
Clean up code

Clean up code

Clean up code within Ultron by removing imports which are not needed

Jh123x committed:
Divide class into packages

Divide class into packages

This is to allow for others and I to find the relavant code more easily

Jh123x committed:
Create ParserTest

Create ParserTest

This is to facilitate unit testing for the Parser Class

Jh123x committed:
Add print method

Add print method

This is to faciliatate refactoring

Jh123x committed:
Removed unused methods

Jh123x committed:
Refactor Code

Refactor code

To reduce code size and improve abstraction

Jh123x committed:
Add DeadLineTest

Add DeadLineTest

Unit test for Deadline class methods

Jh123x committed:
Update .gitignore

Add Manifest file to gitignore

Jh123x committed:
Fix formatting error

Jh123x committed:
Bug fix where done incorrectly shown

Bug fix where done incorrectly shown.

Changed the format of the writer

Jh123x committed:
Change symbol for taskDone

Jh123x committed:
Update coding standard

Jh123x committed:
Renamed Package

Jh123x committed:
Merge pull request #3 from Jh123x/master

Renamed Package

Jh123x committed:
Merge pull request #4 from Jh123x/master

Renamed Package

Jh123x committed:
Update JavaDocs for classes

Jh123x committed:
Add package-javadoc files

Jh123x committed:
Add Find command

Jh123x committed:
Merge pull request #5 from Jh123x/branch-A-JavaDoc

Add JavaDocs

Jh123x committed:
Merge pull request #6 from Jh123x/branch-Level-9

Add Find command

Jh123x committed:
Merge branch 'branch-A-CodingStandard'

Jh123x committed:
Update text-ui-test

Jh123x committed:
Update JavaDocs

Jh123x committed:
Merge pull request #8 from Jh123x/add-gradle-support

Add Gradle support

Jh123x committed:
Bugfix for delete 1

Jh123x committed:
Reduce code length

Jh123x committed:
Update JavaDocs

Jh123x committed:
Update build.gradle to match Ultron

Jh123x committed:
Add Checkstyle.xml from addressbook3

Jh123x committed:
Add supressions.xml from address book 3

Jh123x committed:
Fix Checkstyle Issues

Jh123x committed:
Update CheckStyle

Jh123x committed:
Update CheckStyle

Jh123x committed:
Add JavaFX dependency

Jh123x committed:
Rename Supressions.xml to Suppression.xml

Jh123x committed:
Add Images

Add images to be used for GUI later

Jh123x committed:
Update DeadlineTest

Jh123x committed:
Update build.gradle

Jh123x committed:
Update Messages for ExceptionType

Update Messages for ExceptionType

This is to make it more in line with the Ultron Theme

Jh123x committed:
Add GUI to Ultron

Jh123x committed:
Update CheckStyle

Jh123x committed:
Refactor UI to ui folder

Refactor UI to ui folder

This is to make the code more readable and easier to find.

Jh123x committed:
Update FindCommand.java

Jh123x committed:
Add Unit test for Bye command

Jh123x committed:
Add Unit test for DeleteCommand

Jh123x committed:
Add Unit test for DoneCommands

Jh123x committed:
Bugfix for Find command

Jh123x committed:
Add Unit test for Find Command

Jh123x committed:
Update JavaDocs for FindCommandTest

Jh123x committed:
Add Unit test for Help command

Jh123x committed:
Added Unit test for ListCommand

Jh123x committed:
Update ByeCommand.java

Jh123x committed:
Change way to close application

Jh123x committed:
Update implementation of getResponse

Jh123x committed:
Remove Unused methods

Jh123x committed:
Update Unit test for Deadline

Jh123x committed:
Added Unit test for Event

Jh123x committed:
Add Unit test for Todo

Jh123x committed:
Merge branch 'branch-A-CheckStyle'

Jh123x committed:
Fix UI importing

Jh123x committed:
Update JavaDocs

Update JavaDocs for the project

This is to make sure that the JavaDocs follow the coding style and makes others otherstand what the code is doing.

ysr25 commented on src/main/java/Ultron/Commands/HelpCommand.java:
```diff
@@ -0,0 +1,46 @@
+        }
+
+        ui.setMessage("Heh I guess I could help an insect like you:\n"
+            + "- help                      : Get help for the commands\n"
+            + "- todo (name)               : Adds a todo to the list\n"
+            + "- event (name) /at (date)   : Adds an event at date\n"
+            + "- deadline (name) /by (date): "
+            + "Adds a deadline which expires by date\n"
+            + "- delete (number)           : Removes a todo from the list\n"
```
I think the indentation for these lines should have 4 more spaces. I noticed the same issue in several other places too.

Jh123x replied:
Thank you for pointing it out!

ysr25 reviewed (commented):
Overall nice job, other than the indentation of wrapped lines in some places 👍

AdithyaNarayan commented on src/main/java/Ultron/Commands/Command.java:
```diff
@@ -0,0 +1,56 @@
+     *
+     * @return String arguments.
+     */
+    protected String getArguments() {
```
If argument is a single string, maybe this method could be renamed to `getArgument`?

Jh123x replied:
Yes, that will be a better name. Thank you for pointing it out! :D

AdithyaNarayan commented on src/main/java/Ultron/Commands/FindCommand.java:
```diff
@@ -0,0 +1,54 @@
+        if (getArguments().length() == 0) {
+            throw new UltronException("find", getArguments(), ExceptionType.NO_ARGUMENTS_SUPPLIED);
+        }
+        boolean printed = false;
```
Consider changing to `isPrinted`

Jh123x replied:
Thanks for the advice. Now it sounds more like a boolean name.

AdithyaNarayan commented on src/main/java/Ultron/TaskList.java:
```diff
@@ -0,0 +1,114 @@
+public final class TaskList {
+
+    /** Store the internal list of tasks. */
+    private final ArrayList<Task> list;
```
Consider changing the name to better reflect the intent of the variable. Something along the lines of `tasks`.

Jh123x replied:
Yes, good suggestion :D

AdithyaNarayan reviewed (commented):
A delight to see well-maintained code! Keep it up!

Jh123x committed:
Update storage to use gson

Jh123x committed:
Updated Image to be used

Jh123x committed:
Update DeleteCommand

Update DeleteCommand

Update delete command to use less variables. This will allow the program to take up less space.

Jh123x committed:
Update Assertions

Update Assertions

Update Assertions so that the program can flag it up if there is any critical errors

Jh123x committed:
Fix DeleteCommand

Fix DeleteCommand

Fix DeleteCommand bug where the number of tasks left is not shown correctly

Jh123x committed:
Edit Comments

Jh123x committed:
Revert "Edit Comments"

This reverts commit eb0e17047404d34367a8133ad20dd45be90485f3.

Jh123x committed:
Edit comments

Jh123x committed:
Add ALREADY_DONE exception

Jh123x committed:
Bugfix for DoneCommand

Jh123x committed:
Update FindCommand

Jh123x committed:
Update Application Source

Jh123x committed:
Update Assertions

Jh123x committed:
Merge branch 'branch-A-Assertions'

Jh123x committed:
Update DoneCommand

Jh123x committed:
Removed Redundant methods

Jh123x committed:
Remove WhiteSpace

Jh123x committed:
Remove redundant comments

Jh123x committed:
Remove redundant comments

Jh123x committed:
Remove Redundant comments

Jh123x committed:
Update Deadline

Jh123x committed:
Update Event

Jh123x committed:
Remove Redundant Comments

Jh123x committed:
Remove Redundant Comments

Jh123x committed:
Remove WhiteSpace

Jh123x committed:
Update JavaDocs

Jh123x committed:
Update JavaDocs

Jh123x committed:
Remove Scanner

Jh123x committed:
Update Assertions

Raymond0212 commented on src/main/java/Ultron/Tasks/Todo.java:
```diff
@@ -0,0 +1,57 @@
+package ultron.tasks;
+
+public final class Todo extends Task {
```
Is there a need to declare Todo as a final class? I think there are still some possibilities to extend this class though we might not need it in our iP

Jh123x replied:
When the extension comes, I can just remove the final after that and extend from there?
What I was thinking was that everything extends from the abstract class Task.
If there are other similar classes as todo we can all group them under todo or modify todo to accommodate other functions

Raymond0212 commented on src/main/java/Ultron/Tasks/Event.java:
```diff
@@ -0,0 +1,142 @@
+        Pattern.compile("^(.*) (/at) (.*)$");
+    /** Stores the date format regex.*/
+    private static final DateFormat DATE_FORMAT =
+        new SimpleDateFormat("dd-MM-yyyy HHmm");
```
Would it be simpler if you use LocalDateTime here?

Jh123x replied:
Yes, that might be true. Might edit in a future commit haha. Thanks for letting me know.

Raymond0212 commented on src/main/java/Ultron/Commands/TaskAllocator.java:
```diff
@@ -0,0 +1,87 @@
+
+            //Throw a Duke exception
+            throw new UltronException(command,
+                ExceptionType.INVALID_COMMAND);
```
Indentation for wrapped lines should be 8 spaces.

Jh123x replied:
I see. Thank you!

Raymond0212 reviewed (commented):
Generally, the code quality is super good and the structure of this project is well-organized. In addition, there are tests for each class. I don't have much to say about it. Keep going!

Regards,
Yongyan.

Jh123x committed:
Rename Variable to better names

Jh123x committed:
Changed getArguments to getArgument

Jh123x committed:
Update JavaDocs for TaskCommand

Update JavaDocs for TaskCommand

Use of lambdas present in Enum already

Jh123x committed:
Update FindCommand to use streams

Update FindCommand to use streams

Jh123x committed:
Create gradle.yml

Add workflow for Gradle

Jh123x committed:
Add CSS for styling fxml

Jh123x committed:
Reformat Code

Jh123x committed:
Remove JavaDocs

Jh123x committed:
Adjust line spacing

Jh123x committed:
Update help message

Update help message to show find

Jh123x committed:
Update Acknowledgement

Jh123x committed:
Update Readme

Jh123x committed:
Update build.gradle

Jh123x committed:
Add UI Image

Jh123x committed:
Removed UI.png

Jh123x committed:
Add Ui.png

Jh123x committed:
Refactor Parser

Jh123x committed:
Merge pull request #9 from Jh123x/Refactor-Parser

Refactor Parser

Jh123x committed:
Make variable final
